src/plugins/orangehrmMaintenancePlugin/PurgeStrategy/DestroyCandidatePurgeStrategy.php

```diff
@@ -0,0 +1,40 @@
+<?php
+/**
+ * OrangeHRM is a comprehensive Human Resource Management (HRM) System that captures
+ * all the essential functionalities required for any enterprise.
+ * Copyright (C) 2006 OrangeHRM Inc., http://www.orangehrm.com
+ *
+ * OrangeHRM is free software; you can redistribute it and/or modify it under the terms of
+ * the GNU General Public License as published by the Free Software Foundation; either
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Maintenance\PurgeStrategy;
+
+use OrangeHRM\Entity\Candidate;
+
+class DestroyCandidatePurgeStrategy extends PurgeStrategy
+{
+    /**
+     * @param int $vacancyNumber
+     */
+    public function purge(int $vacancyNumber): void
+    {
+        $matchByValues = $this->getMatchByValues($vacancyNumber);
+        $purgeEntities = $this->getEntityRecords($matchByValues, $this->getEntityClassName());
+        foreach ($purgeEntities as $purgeEntity) {
+            /** @var Candidate $purgeEntity */
+            if (!$purgeEntity->isConsentToKeepData()) {
+                $this->getEntityManager()->remove($purgeEntity);
+            }
+        }
```

Comment:
1. ref: https://www.doctrine-project.org/projects/doctrine-orm/en/2.11/reference/batch-processing.html#bulk-deletes
2. or we can go for query builder,

```php
        $qb = $this->createQueryBuilder(Candidate::class, 'candidate');
        $qb->leftJoin('candidate.candidateVacancy', 'candidateVacancy');
        $qb->delete()
            ->andWhere('candidate.consentToKeepData = :consentToKeepData')
            ->setParameter('consentToKeepData', false)
            ->andWhere('candidateVacancy.vacancy = :vacancyId')
            ->setParameter('vacancyId', $vacancyId);
```
not sure about the 2nd approach with the design

Reply:
Is what I've done okay? @RajithaKumara